File: SDKStydy/StudySDK/StudySDKDemo/StudySDKDemo/ViewController.m
//
//  ViewController.m
//  StudySDKDemo
//
//  Created by tzsoft on 2018/6/14.
//  Copyright © 2018年 HLY. All rights reserved.
//

#import "ViewController.h"
#import <StudySDK/SDKView.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    SDKView *testView = [[SDKView alloc]initWithColor:[UIColor orangeColor]];
    testView.frame = CGRectMake(0, 80, 200, 200);
    [self.view addSubview:testView];
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Swift/基本语法/第二部分/11-自定义Log/11-自定义Log/ViewController.swift
//
//  ViewController.swift
//  11-自定义Log
//
//  Created by tzsoft on 2018/2/26.
//  Copyright © 2018年 胡小羊. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        // 1 获取打印所在的文件
        let file = #file
        print("\(file)")
        
        // 2 获取打印所在的方法
        let funName = #function
        print("\(funName)")
        
        // 3 获取打印所在的行数
        let lineNum = #line
        print("\(lineNum)")
        
        DLog(message: "自定义log")
        DLog(message: "hahahahaahha")
    }
}

